spec/controllers/admin/customers_controller_spec.rb

```diff
@@ -164,19 +164,27 @@ def expect_locals
       customer = create(:customer)
 
       expect do
-        delete :destroy, {id: customer.to_param}
+        delete :destroy, kwarg_params(id: customer.to_param)
       end.to change(Customer, :count).by(-1)
     end
 
     it "redirects to the customers list" do
       customer = create(:customer)
 
-      delete :destroy, id: customer.to_param
+      delete :destroy, kwarg_params(id: customer.to_param)
 
       expect(response).to redirect_to(admin_customers_url)
     end
   end
 
+  def kwarg_params(args)
+    if Rails::VERSION::MAJOR >= 5
+      {params: args}
+    else
+      args
+    end
```

Comment:
Actually this is the worst place to put in any kind of optimal code. This is a yak shaving method to bring about compatibility in a situation where the api has broken. 

It will eventually be removed, but till then it should be as self documenting as possible and as plain spoken as possible lest it turn into an actual valuable feature. 

347.433.0567 | @imajes 

> On May 24, 2016, at 22:49, Ben A. Morgan <notifications@github.com> wrote:
> 
> In spec/controllers/admin/customers_controller_spec.rb:
> 
> >  
> >        expect(response).to redirect_to(admin_customers_url)
> >      end
> >    end
> >  
> > +  def kwarg_params(args)
> > +    if Rails::VERSION::MAJOR >= 5
> > +      {params: args}
> > +    else
> > +      args
> > +    end
> Yep, that's another solution. I would opt for the ternary mainly because I think the purpose of the method is to format the params if its greater than or equal to Rails 4... If we had to use guard clauses, I think a more optimal outcome would be:
> 
> return args if Rails::VERSION::MAJOR <= 4
> { params: args }
> 🚲 🏠
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
>